usecases/evaluate_scenario/evaluate_scenario.go

```diff
@@ -114,9 +124,35 @@ func EvalScenario(
 		return models.ScenarioExecution{}, err
 	}
 
+	var pivotValue *string
+	if params.Pivot != nil {
+		pivotValue, err = getPivotValue(ctx, *params.Pivot, dataAccessor)
+		if err != nil {
+			return models.ScenarioExecution{}, errors.Wrap(
+				err,
+				"error getting pivot value in EvalScenario")
+		}
+	}
+
+	snoozes := make([]models.RuleSnooze, 0)
+	if params.Pivot != nil {
+		snoozeGroupIds := make([]string, 0, len(publishedVersion.Body.Rules))
+		for _, rule := range publishedVersion.Body.Rules {
+			if rule.SnoozeGroupId != nil {
+				snoozeGroupIds = append(snoozeGroupIds, *rule.SnoozeGroupId)
+			}
+		}
+		snoozes, err = repositories.SnoozeReader.ListRuleSnoozesForDecision(ctx, exec, snoozeGroupIds, *pivotValue)
+	}
```

Comment:
I would do one of the following instead (IIUC, the pivot value can be null, so snooze if statement should depend on it):
```suggestion
	var pivotValue *string
	if params.Pivot != nil {
		pivotValue, err = getPivotValue(ctx, *params.Pivot, dataAccessor)
		if err != nil {
			return models.ScenarioExecution{}, errors.Wrap(
				err,
				"error getting pivot value in EvalScenario")
		}
	}

	snoozes := make([]models.RuleSnooze, 0)
	if pivotValue != nil {
		snoozeGroupIds := make([]string, 0, len(publishedVersion.Body.Rules))
		for _, rule := range publishedVersion.Body.Rules {
			if rule.SnoozeGroupId != nil {
				snoozeGroupIds = append(snoozeGroupIds, *rule.SnoozeGroupId)
			}
		}
		snoozes, err = repositories.SnoozeReader.ListRuleSnoozesForDecision(ctx, exec, snoozeGroupIds, *pivotValue) // ensure *pivotValue won't fail (independant from a change in above if statement
	}
```

Reply:
Thank you, your suggestion was totally my intent, I failed at writing line 138 (and didn't properly test this case locally)